analyzer/diagnostics.go

```diff
@@ -0,0 +1,26 @@
+package analyzer
+
+import (
+	"golang.org/x/tools/go/analysis"
+)
+
+func newAnalysisDiagnostic(
+	checker string,
+	analysisRange analysis.Range,
+	message string,
+	suggestedFixes []analysis.SuggestedFix,
+) *analysis.Diagnostic {
+	d := analysis.Diagnostic{
+		Pos:            analysisRange.Pos(),
+		End:            analysisRange.End(),
+		Message:        message,
+		SuggestedFixes: suggestedFixes,
+	}
+	if len(checker) > 0 {
```

Comment:
```suggestion
	if checker != "" {
```

https://dmitri.shuralyov.com/idiomatic-go#empty-string-check

Reply:
There is a gocritic rule for this one

https://github.com/go-critic/go-critic/blob/master/docs%2Foverview.md

emptyStringLen